### ISSUE TITLE
oxfmt/jsdoc: Issue with code fence in @example (indent, ASI)

### ISSUE THREAD

**Issue by Mister-Hope** [oxc-project/oxc#21549]
A jsdoc example here:

```js
  /**
   * xxx
   *
   * @example
   * ```html
   * <a title="a">text</a>
   * ```
   */
```

VSCode can render it correctly:

<img width="594" height="114" alt="Image" src="https://github.com/user-attachments/assets/dd07576c-dfda-4916-b67c-d32ac04d6b33" />

But oxfmt refuses it and forces it to be a JS code snippet:

```
  /**
   * Xxx
   *
   * @example
   *   ;```html
   *   <a title="a">text</a>
   *   ```
   */
```

**Comment by leaysgur** [comment]
```
❯ cat a.ts
  /**
   * xxx
   *
   * @example
   * ```html
   * <a title="a">text</a>
   * ```
   */
  const x =1


❯ cat cfg.json
{ "jsdoc": true }

❯ npx oxfmt@0.45.0 --config cfg.json a.ts
Finished in 9ms on 1 files using 8 threads.

❯ cat a.ts
/**
 * Xxx
 *
 * @example
 *   ```html
 *   <a title="a">text</a>
 *   ```
 */
const x = 1;
```

Could you please explain in detail what the issue is and what the expected result should be? (or, minimal repro)

**Comment by Mister-Hope** [comment]
Out team use `semi: false`

The code fence still being parsed as javascript, so when it found a `` ` ``, it adds a `;` at line start.

<img width="343" height="254" alt="Image" src="https://github.com/user-attachments/assets/961aef7f-d33e-467a-953e-9cf286b8d7ec" />

<img width="415" height="214" alt="Image" src="https://github.com/user-attachments/assets/5b43c4ce-e0fb-4e4a-a15b-c78d2dcbefa9" />

We want: 

```js
/**
 * Xxx
 *
 * @example
 *
 * ```html
 * <a title="a">text</a>
 * ```
 */
export type Xxx = string
```

(Note: the extra line here does not matter, oxfmt removes it anyway)

Render result:

<img width="419" height="218" alt="Image" src="https://github.com/user-attachments/assets/883033d7-b3f4-4498-9b0e-36217950e6bb" />

The code fence should be never indented at all. If indented with 2 spaces, it already becomes javascript and loses highlighting:

<img width="362" height="235" alt="Image" src="https://github.com/user-attachments/assets/6297ea61-e5d7-4dc9-b94d-e99e27b6bb23" />

So your `a.ts` already breaks, even without `semi: false`.

**Comment by leaysgur** [comment]
```
❯ cat a.ts
/**
 *
 * Xxx
*
 * @example
 *   ```html
 *   <a title="a">text</a>
 *   ```
 */
export type Xxx=string;

❯ cat cfg.json
{ "semi": false, "jsdoc": true }

❯ npx oxfmt@0.45.0 --config cfg.json a.ts
Finished in 10ms on 1 files using 8 threads.

❯ cat a.ts
/**
 * Xxx
 *
 * @example
 *   ;```html
 *   <a title="a">text</a>
 *   ```
 */
export type Xxx = string
```

Confirmed.

**Comment by Mister-Hope** [comment]
Hi, just a friendly ping, I updated my original comment, your original one (without `semi: false` also breaks), see screenshot. 2 indents shall never be applied anyway.

**Comment by skoblenick** [comment]
This bug is a side-effect of a different issue. I have opened an PR here: https://github.com/oxc-project/oxc/pull/21687 Which should address the root cause.

**Comment by leaysgur** [comment]
I took some time to investigate..., so let me confirm my current understanding.

There are two issues reported here:

1. Unexpected `;` being inserted before(semi:false) / after(semi:true) the fence marker
2. Fenced code blocks being indented (2 spaces)

And our current implementation produces the same output as `prettier-plugin-jsdoc` for both cases. In that sense, we are correctly matching the upstream behavior.

However, it's worth noting that `prettier-plugin-jsdoc` does not intentionally support fenced code blocks inside `@example`. It doesn't recognize them as fenced blocks at all.

https://github.com/fardad-dev/prettier-plugin-jsdoc/issues/241

Anyway, we already diverge from upstream on several behaviors, so improving the fenced code block handling is also possible, I think.

https://github.com/oxc-project/oxc/blob/main/tasks/prettier_conformance/jsdoc/upstream-jsdoc-bugs.md

---

Update: VS Code (TS extension) renders `@example` as Markdown if the body contains a fence (`/^\s*[~`]{3}/m`), otherwise wraps it in a code block.
So indenting fences breaks rendering (4+ spaces = indented code block in CommonMark, see #24729).

Proposed direction, mirroring that heuristic:

- Body contains a fence → markdown mode: no indent, format JS/TS fence contents, keep others verbatim, never parse the whole body as JS (also fixes the ASI `;`)
- No fence → keep current upstream-compatible behavior (VS Code wraps it in a code block anyway)
- Accept and dedent indented fences on input, so previously broken output self-heals
- Upstream doesn't support fences in `@example` (fardad-dev/prettier-plugin-jsdoc#241), so this is another intentional divergence

**Comment by R-Bower** [comment]
This is a significant issue that is preventing me from fully migrating to oxlint / oxfmt from ESLint and prettier.
